### DISCUSSION TITLE
[Accessibility] In-engine documentation: Distinguish different fonts better in help descriptions of code components

### DISCUSSION THREAD

**Discussion by clockworkprince** [godotengine/godot-proposals#4804] [Editor]
(This might be in the wrong category; I couldn't find one for documentation specifically, but this is related to the help that's available in the editor itself.)

I've been struggling to comprehend some of the documentation provided within Godot itself for code components. It's only just now that I realised that some of the words in the documentation had been written in a different, yet very similar font compared to the rest of the description.

An example of this:
![image](https://user-images.githubusercontent.com/47702744/177059281-a8d7922d-f8af-42ff-a702-8d155c455b59.png)

My issue with this is that the fonts look so similar to me that I have a hard time telling when one is used over the other, especially for such short words as "to" (which is seen in the above example). I do not have dyslexia, but if I struggle with distinguishing these two fonts, someone who _does_ have dyslexia might struggle a lot more than me. 

Some suggestions to fix this issue: 
- Change either font so they are easier to recognise as different fonts. 
- Use highlighting functionality to apply a darker colour highlight around the monospace font.
- Apply a different font colour to the monospace font. 
- Allow editing of the used font, highlighting colour and font colours for the help documentation in the Editor Settings.

Having all of these applied would be really nice, but even if just half of them get implemented I'd already be very happy.

**Comment by Calinou** [comment]
We can add a background color for inline code now that we have [`[bgcolor]` in RichTextLabel](https://github.com/godotengine/godot/pull/35608) in `master`. However, `[bgcolor]` isn't available in `3.x`'s RichTextLabel, so this can only be implemented in 4.0.

Doing this for multiline code blocks may prove more difficult though, especially if you want the actual *block* to be highlighted, rather than several line highlights.

> - Allow editing of the used font, highlighting colour and font colours for the help documentation in the Editor Settings.

You can change the main and code fonts in the editor settings to your liking. This will also affect the class reference.

The highlighting color is derived from the accent color set in the editor theme, which can also be changed in the editor settings.

**Reply by clockworkprince** [reply]
> We can add a background color for inline code now that we have [`[bgcolor]` in RichTextLabel](https://github.com/godotengine/godot/pull/35608) in `master`. However, `[bgcolor]` isn't available in `3.x`'s RichTextLabel, so this can only be implemented in 4.0.

That would be very nice! I've been looking forward to Godot 4.0 in general, though I'm not sure what the expected timeline for a stable release is, so I've been holding off on trying it out. (I'm fairly new to Godot anyway, so I'd have issues with not having much in the way of tutorials or documentation.)

> Doing this for multiline code blocks may prove more difficult though, especially if you want the actual _block_ to be highlighted, rather than several line highlights.

That's alright. It's mainly for inline code snippets that this is important for me. I need to be able to visually distinguish where there is code within sentences in the explanations for it, basically. I don't mind blocks of just code not being as visually distinct, because they are easier to recognise for me personally as code instead of English or another non-code language.  

> > * Allow editing of the used font, highlighting colour and font colours for the help documentation in the Editor Settings.
> 
> You can change the main and code fonts in the editor settings to your liking. This will also affect the class reference.

I had looked for this before, but struggled to find it. Somehow, I managed to find it this time around. I've immediately changed the fonts to something that is easier to distinguish for me. Thanks for pointing this out!

> The highlighting color is derived from the accent color set in the editor theme, which can also be changed in the editor settings.

That seems to be for selection-based highlighting, which isn't what I was talking about. I meant more automatic high-lighting that's not done via user input, such as how Discord tends to give a different highlight colour around code:

![image](https://user-images.githubusercontent.com/47702744/177161589-987c680e-cdb8-4348-9fad-9f5aa902f7ca.png)

This is basically something I would like to have in the in-editor Godot documentation too.

**Reply by Calinou** [reply]
> That seems to be for selection-based highlighting, which isn't what I was talking about. I meant more automatic high-lighting that's not done via user input, such as how Discord tends to give a different highlight colour around code:

I just checked, the code color *is* derived from the accent color. It's blended 60% between the accent color and base color (pure white for dark themes, pure black for light themes): https://github.com/godotengine/godot/blob/41fdddb1f8132dcba8256a145de44da5df3155b2/editor/editor_themes.cpp#L1409

**Comment by Calinou** [comment]
I opened a pull request improving highlighting of code, keyboard shortcuts and clickable references: https://github.com/godotengine/godot/pull/62710
